index.js

```diff
@@ -52,14 +52,17 @@ async function getTokensOfOwner(
   wantMetadata = true
 ) {
   const contract = new web3.eth.Contract(ERC721ABI, contractAddress);
-  const total = await contract.methods.balanceOf(ownerAddress).call();
+  const total = parseInt(await contract.methods.balanceOf(ownerAddress).call(), 10);
   const tokens = [];
 
   return new Promise((resolve, reject) => {
+    if (total === 0) {
```

Comment:
This is what is actually fixing #1. Rest just cleans up. Feel free to just take that chunk.